django-rgd-3d/rgd_3d/tasks/etl.py

```diff
@@ -57,24 +54,27 @@ def _get_readers():
     return methods
 
 
-def read_point_cloud_file(pc_file):
-    """Read a PointCloud object and create a new PointCloudMeta."""
-    if not isinstance(pc_file, PointCloud):
-        pc_file = PointCloud.objects.get(pk=pc_file)
-    pc_entry, _ = get_or_create_no_commit(PointCloudMeta, source=pc_file)
-    pc_entry.name = pc_file.file.name
+def read_mesh_3d_file(mesh_3d):
+    """Read a Mesh3D object and populate the vtp_data field."""
+    if not isinstance(mesh_3d, Mesh3D):
+        mesh_3d = Mesh3D.objects.get(pk=mesh_3d)
+    if not mesh_3d.name:
+        mesh_3d.name = mesh_3d.file.name
     # Parse the point cloud file format and convert
-    ext = pc_file.file.name.split('.')[-1].strip().lower()
+    ext = mesh_3d.file.name.split('.')[-1].strip().lower()
     try:
         method = _get_readers()[ext]
     except KeyError:
         raise ValueError(f'Extension `{ext}` unsupported for point cloud conversion.')
-    try:
-        pc_entry.vtp_data
-    except ChecksumFile.DoesNotExist:
-        pc_entry.vtp_data = ChecksumFile()
-    if not pc_entry.vtp_data.collection:
-        pc_entry.vtp_data.collection = pc_file.file.collection
-    _file_conversion_helper(pc_file.file, pc_entry.vtp_data.file, method, extension='.vtp')
+    if not mesh_3d.vtp_data:
+        mesh_3d.vtp_data = ChecksumFile()
+    if not mesh_3d.vtp_data.collection:
+        mesh_3d.vtp_data.collection = mesh_3d.file.collection
```

Comment:
Eventually we ought to move this type of validation out of ETL and into the [`Model.clean()`](https://docs.djangoproject.com/en/3.2/ref/models/instances/#django.db.models.Model.clean) method so other tasks can benefit from it. However, this is the pattern we use elsewhere, so I think this is good for now.